libraries/audUtils/bidderUtils.js

```diff
@@ -0,0 +1,129 @@
+import {
+  deepSetValue,
+  generateUUID,
+  logError
+} from '../../src/utils.js';
+
+// Function to get Request
+export const getBannerRequest = (bidRequests, bidderRequest, ENDPOINT) => {
+  let request = [];
+  // Loop for each bid request
+  bidRequests.forEach(bidReq => {
+    let guid = generateUUID();
+    const req = {
+      id: guid,
+      imp: [getImpDetails(bidReq)],
+      placementId: bidReq.params.placement_id,
+      site: getSiteDetails(bidderRequest),
+      user: getUserDetails(bidReq)
+    };
+    // Fetch GPP Consent from bidderRequest
+    if (bidderRequest && bidderRequest.gppConsent && bidderRequest.gppConsent.gppString) {
+      deepSetValue(req, 'regs.gpp', bidderRequest.gppConsent.gppString);
+      deepSetValue(req, 'regs.gpp_sid', bidderRequest.gppConsent.applicableSections);
+    } else if (bidderRequest && bidderRequest.ortb2 && bidderRequest.ortb2.regs && bidderRequest.ortb2.regs.gpp) {
+      deepSetValue(req, 'regs.gpp', bidderRequest.ortb2.regs.gpp);
+      deepSetValue(req, 'regs.gpp_sid', bidderRequest.ortb2.regs.gpp_sid);
+    }
+    // Fetch coppa compliance from bidderRequest
+    if (bidderRequest && bidderRequest.ortb2 && bidderRequest.ortb2.regs && bidderRequest.ortb2.regs.coppa) {
+      deepSetValue(req, 'regs.coppa', 1);
+    }
+    // Fetch uspConsent from bidderRequest
+    if (bidderRequest?.uspConsent) {
+      deepSetValue(req, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    }
+    request.push(req);
+  });
+  // Return the array of request
+  return {
+    method: 'POST',
+    url: ENDPOINT,
+    data: JSON.stringify(request),
+    options: {
+      contentType: 'application/json',
+    }
+  };
+}
+// Function to get Response
+export const getBannerResponse = (bidResponse, mediaType) => {
+  let responseArray = [];
+  if (bidResponse) {
+    try {
+      let bidResp = bidResponse.body && bidResponse.body.seatbid && bidResponse.body.seatbid[0] ? bidResponse.body.seatbid[0].bid : [];
```

Comment:
This would be better handled with deepAccess and looping through the result:

`let bidResp = deepAccess(bidResponse, 'body.seatbid', [])`